docs/source/howto/codes.rst

```diff
@@ -3,173 +3,199 @@
+1. Prepare the required input files
+2. Run the code with the correct command line parameters
 
-.. todo::
+Finally, you will probably want a *parser* plugin, which tells AiiDA how to:
+
+3. Parse the output of the code.
 
-    Add to preceding sentence: :ref:`the communication with external machines<how-to:codes:transport>` and the interaction with its :ref:`scheduling software<how-to:codes:scheduler>`.
 
-Some general guidelines to keep in mind are:
-
- * | **Check existing resources.**
-   | Before starting to write a plugin, check on the `aiida plugin registry <https://aiidateam.github.io/aiida-registry/>`_ whether a plugin for your code is already available.
-     If it is, there is maybe no need to write your own, and you can skip straight ahead to :ref:`running the code<how-to:codes:run>`.
- * | **Start simple.**
-   | Make use of existing classes like :py:class:`~aiida.orm.nodes.data.dict.Dict`, :py:class:`~aiida.orm.nodes.data.singlefile.SinglefileData`, ...
-     Write only what is necessary to pass information from and to AiiDA.
- * | **Don't break data provenance.**
-   | Store *at least* what is needed for full reproducibility.
- * | **Expose the full functionality.**
-   | Standardization is good but don't artificially limit the power of a code you are wrapping - or your users will get frustrated.
-     If the code can do it, there should be *some* way to do it with your plugin.
- * | **Don't rely on AiiDA internals.**
-     Functionality at deeper nesting levels is not considered part of the public API and may change between minor AiiDA releases, breaking your plugin.
- * | **Parse what you want to query for.**
-   | Make a list of which information to:
-
-     #. parse into the database for querying (:py:class:`~aiida.orm.nodes.data.dict.Dict`, ...)
-     #. store in the file repository for safe-keeping (:py:class:`~aiida.orm.nodes.data.singlefile.SinglefileData`, ...)
-     #. leave on the computer where the calculation ran (:py:class:`~aiida.orm.nodes.data.remote.RemoteData`, ...)
-
-To demonstrate how to create a plugin for an external code, we will use the trivial example of using the `bash` shell (``/bin/bash``) to sum two numbers by running the command: ``echo $(( numx + numy ))``.
-Here, the `bash` binary will be effectively acting as our |Code| executable, the input (``aiida.in``) will then be a file containing the command with the numbers provided by the user replaced, and the output (``aiida.out``) will be caught through the standard output.
-The final recipe to run this code will then be:
+This how-to takes you through the process of :ref:`creating a calculation plugin<how-to:codes:interfacing>` for a dummy executable that sums two numbers, using it to :ref:`run the code<how-to:codes:run>`, and :ref:`writing a parser <how-to:codes:parsing>` for its outputs.
```

Comment:
```suggestion
This how-to takes you through the process of :ref:`creating a calculation plugin<how-to:codes:interfacing>` for a very simple executable that sums two numbers, using it to :ref:`run the code<how-to:codes:run>`, and :ref:`writing a parser <how-to:codes:parsing>` for its outputs.
```
It is not actually a dummy executable since it performs the desired calculation.